nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java

```diff
@@ -362,6 +377,23 @@
             .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
             .build();
 
+    public static final PropertyDescriptor SSL_CONTEXT_SERVICE = new PropertyDescriptor.Builder()
+            .name("SSL Context Service")
+            .displayName("SSL Context Service")
+            .description("SSL Context Service supporting encrypted socket communication")
+            .required(false)
+            .identifiesControllerService(SSLContextService.class)
+            .build();
+
+    public static final PropertyDescriptor SSL_MODE = new PropertyDescriptor.Builder()
+            .name("SSL Mode")
+            .displayName("SSL Mode")
+            .description("SSL Mode used when SSL Context Service configured supporting certificate verification options")
+            .required(true)
+            .defaultValue(SSLMode.DISABLED.toString())
+            .allowableValues(SSL_MODES)
+            .build();
+
```

Comment:
Although there is some similarity, the SSLMode options control the flexibility of the MySQL Binary Log client during TLS negotiation, whereas ClientAuth influences how an SSL Server Socket would respond.  Since the Binary Log connection is a client, the ClientAuth options are not necessarily applicable.

SSLMode provides some more lenient connection options, such as PREFERRED, where the Binary Log client will request SSL, but will still connect and function if the MySQL server does not support SSL.  The VERIFY_IDENTITY option is a special case that causes the Binary Log client to use a custom TlsHostnameVerifier to compare the network connected hostname to the certificate presented.  The REQUIRED and VERIFY_CA options result in essentially the same behavior for the purposes of CaptureChangeMySQL since SSLContextService always provides the SSLSocketFactory instead of having the Binary Log client use it's own preconfigured factories.

Another option would be attempting to translate ClientAuth values to an applicable SSLMode, but there isn't a direct correlation, and I think that would end up being more confusing.  Based on how the Binary Log client uses SSLMode, it might make sense to limit the allowable values to DISABLED, PREFERRED, REQUIRED, and VERIFY_IDENTITY.  As mentioned, the VERIFY_IDENTITY option is a special case that would be more restrictive than REQUIRED due to the custom hostname verification.  If that sounds too confusing, then it would be best to simply remove the SSL Mode property and use the presence of an SSLContextService to set REQUIRED.

Reply:
Yeah very good points, it makes sense. Can we add the comments from the library to the description of the allowable values? you will have to duplicate the enum on NiFi's side I'm afraid but will provide more clarity to our users. Thoughts?

Reply:
Thanks for the feedback.  Adding more description around each allowable value makes sense since the options or not necessarily intuitively obvious.  Based on how CaptureChangeMySQL will provide the SSLSocketFactory, I will also remove VERIFY_CA as an option.

Reply:
@pvillard31 Thanks again for the initial feedback, do you have any other recommendations following the change to include a detailed description of the SSL Mode properties?